workflow-manager/utils/utils.go

```diff
@@ -8,3 +22,62 @@ func Index(isFirst bool) int {
 	}
 	return 1
 }
+
+// Clock allows mocking of time for testing purposes
+type Clock struct {
+	now time.Time
+}
+
+// DefaultClock reutrns a Clock that returns the real time from time.Now
+func DefaultClock() Clock {
+	return Clock{}
+}
+
+// ClockWithFixedNow returns a Clock that always returns the provided now
+func ClockWithFixedNow(now time.Time) Clock {
+	return Clock{now: now}
+}
+
+// Now returns the current time according to this clock
+func (c *Clock) Now() time.Time {
+	if c.now.IsZero() {
+		return time.Now()
+	}
+	return c.now
+}
```

Comment:
This approach is also implemented by https://github.com/jmhodges/clock, in case you would prefer to use that.

Reply:
Had I known, I would have used it, but I'm happy with this `Clock`.